app/temporal/src/domain/DataAnalyzer.ts

```diff
@@ -52,6 +57,75 @@ export class DataAnalyzer {
     return [...foundColumnRecommendations, ...notFoundMatchRecommendations];
   }
 
+  public processDataValidations(
+    data: Record<string, unknown>[],
+    validatorColumns: ValidatorColumns,
+    stats: Stats
+  ): { rowId: number; column: string; errors: ValidationError[] }[] {
+    const chunkErrors: {
+      rowId: number;
+      column: string;
+      errors: ValidationError[];
+    }[] = [];
+    for (const row of data) {
+      for (const validatorKey of Object.keys(
+        validatorColumns
+      ) as ValidatorType[]) {
+        if (validatorColumns[validatorKey].length > 0) {
+          const errors = validators[validatorKey].validate(
+            row,
+            validatorColumns[validatorKey],
+            stats
+          );
+
+          for (const column of Object.keys(errors as any)) {
+            const error = errors[column] as any;
+            const errorForRowAndColumn = chunkErrors.find(
+              (item) => item.rowId === row.__rowId && item.column === column
+            );
+            if (errorForRowAndColumn) {
+              errorForRowAndColumn.errors.push(error);
+            } else {
+              chunkErrors.push({
+                rowId: row.__rowId as number,
+                column,
+                errors: [error],
+              });
+            }
+          }
+        }
+      }
+    }
+    return chunkErrors;
+  }
+
+  public getStats(
+    data: Record<string, unknown>[],
+    columnsToVerify: string[]
+  ): Stats {
+    // nonunique
+    const stats = {} as Stats;
+    for (const column of columnsToVerify) {
+      const columnValues = data.map((row) => row[column]);
```

Comment:
Ich glaube es wäre bessern, wenn man nicht erst die columnValues in einem Array speichert, sondern gleich in einer Map und beim durchgehen prüft existiert der Wert => count + 1 ansonsten add Wert mit 1
Dann im zweiten die Map nochmal durchgehen und alle Werte mit count = 1 entfernen